pkg/mongodb/Symfony/MongodbTransportFactory.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Enqueue\Mongodb\Symfony;
+
+use Enqueue\Mongodb\Client\MongodbDriver;
+use Enqueue\Mongodb\MongodbConnectionFactory;
+use Enqueue\Mongodb\MongodbContext;
+use Enqueue\Symfony\DriverFactoryInterface;
+use Enqueue\Symfony\TransportFactoryInterface;
+use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+class MongodbTransportFactory implements TransportFactoryInterface, DriverFactoryInterface
+{
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @param string $name
+     */
+    public function __construct($name = 'Mongodb')
+    {
+        $this->name = $name;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addConfiguration(ArrayNodeDefinition $builder)
+    {
+        $builder
+            ->beforeNormalization()
+                ->ifString()
+                ->then(function ($v) {
+                    return ['dsn' => $v];
+                })
+            ->end()
+            ->children()
+                ->scalarNode('dsn')
+                    ->info('The Mongodb DSN. Other parameters are ignored if set')
+                ->end()
+                ->scalarNode('dbname')
+                    ->defaultValue('enqueue')
+                    ->info('Database name.')
+                ->end()
+                ->scalarNode('collection_name')
+                    ->defaultValue('enqueue')
+                    ->info('Collection')
+                ->end()
+                ->integerNode('polling_interval')
+                    ->defaultValue(1000)
+                    ->min(100)
+                    ->info('How often query for new messages.')
+                ->end()
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createConnectionFactory(ContainerBuilder $container, array $config)
+    {
+        if (false == empty($config['dsn'])) {
+            $factory = new Definition(MongodbConnectionFactory::class);
+            $factory->setArguments([$config]);
```

Comment:
as far as I remember Definition takes an array of arguments as a second constructor argument.